.github/workflows/driver-schema-version.yml

```diff
@@ -1,36 +1,48 @@
-# NOTE: it is UNSAFE to run ANY kind of script when using the pull_request_target trigger!
-# DO NOT TOUCH THIS FILE UNLESS THE TRIGGER IS CHANGED.
-# See warning in https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target.
-
 name: Driver SCHEMA_VERSION checks
 on:
-  pull_request_target:
-    paths:
-      - 'driver/event_table.c'
+  pull_request:
+  push:
+    branches:
+      - master
+      - 'release/**'
+
+# Checks if any concurrent jobs under the same pull request or branch are being executed
+concurrency:
+  group: driver-schema-version-${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
 
 jobs:
   paths-filter:
     runs-on: ubuntu-latest
     outputs:
-      driver_schema_changed: ${{ steps.filter.outputs.driver_schema }}
+      schema_version_changed: ${{ steps.filter.outputs.schema_version }}
+      schema_defs_changed: ${{ steps.filter.outputs.schema_defs }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - uses: dorny/paths-filter@v2
       id: filter
       with:
         filters: |
-          driver_schema:
+          schema_version:
             - 'driver/SCHEMA_VERSION'
+          schema_defs:
+            - 'driver/event_table.c'
+            - 'driver/ppm_events_public.h'
+            - 'driver/event_stats.h'
+            - 'driver/syscall_*'
 
-  driver-schema:
+  check-version-bump:
     runs-on: ubuntu-22.04
-    permissions:
-      pull-requests: write
     needs: paths-filter
-    if: needs.paths-filter.outputs.driver_schema_changed == 'false'
+    if: needs.paths-filter.outputs.schema_defs_changed == 'true' && needs.paths-filter.outputs.schema_version_changed == 'false'
     steps:
+      - uses: actions/checkout@v3
+
       - name: Check driver SCHEMA_VERSION
-        uses: mshick/add-pr-comment@v2
-        with:
-          message: |
-            Please double check **driver/SCHEMA_VERSION** file. See [versioning](https://github.com/falcosecurity/libs/blob/master/driver/README.VERSION.md#schema-version-number).
+        run: |
```

Comment:
So, are we just failing the check? Without any comment nor further notice to the user?

Reply:
I mean, i don't like the fact that the API_VERSION check is posting a comment, while this check is going to fail.

Reply:
IMO a comment is useful when we have a non-failing silent warning, because nobody would read the message if it was simply logged in a successful CI job. On the other hand here we're turning it into an actual check that can lead to either a failure be skipped, meaning that the output is clearly visibile by everyone. In case the job fails, reviewers and contributors can check the message and understand that versioning is the issue.

As for consistency, I agree with you that it's ugly to have two different approaches. But for a matter of purpose, I think this makes sense. I'm not gonna fight for this though, it's just my opinion and I'm open to change.

Reply:
I mean, what if i just add a comment to driver/event_table?
Would i need to forcefully bump the SCHEMA_VERSION just to silence the CI check?